src/main/java/xyz/niflheim/stockfish/engine/UCIEngine.java

```diff
@@ -92,7 +105,22 @@ private void passOption(Option option) {
     }
 
     private String getPath(Variant variant, String override) {
-        StringBuilder path = new StringBuilder(override == null ? "assets/engines/stockfish_10_x64" : override + "stockfish_10_x64");
+        return getPath(variant, override, null);
+    }
+
+    private String getPath(Variant variant, String override, Integer engineVersion) {
+        Optional<Integer> versionOptional = SUPPORTED_VERSIONS.values().stream().filter(v -> v == engineVersion).findFirst();
```

Comment:
I don't think stream is necessary here. By changing the map signature, this code can be simplified